recipes/cc65/all/conanfile.py

```diff
@@ -0,0 +1,152 @@
+from conans import AutoToolsBuildEnvironment, ConanFile, MSBuild, tools
+from conans.errors import ConanInvalidConfiguration
+from contextlib import contextmanager
+import os
+
+
+class Cc65Conan(ConanFile):
+    name = "cc65"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://cc65.github.io/"
+    description = "A freeware C compiler for 6502 based systems"
+    license = "zlib"
+    topics = ("conan", "cc65", "compiler", "cmos", "6502", "8bit")
+
+    settings = "os_build", "arch_build", "compiler", "arch"
+
+    _autotools = None
+    _source_subfolder = "source_subfolder"
+
+    def configure(self):
+        # MSBuild needs a arch attribute. Make sure arch_build and arch are the same.
+        self.settings.arch = self.settings.arch_build
```

Comment:
```suggestion
```

Overriding settings is not allowed in the official recipes. You can raise an error when cross-building x86 instead.

Reply:
I understand, but this recipe is only packaging a cc65 compiler.
What I want to do is just compile it under Visual Studio as x86.
x86_64 and x86 are compatible. Because it does not create any libraries that must be linked to, it does not matter.

Reply:
Also, if settings is defined as `arch_build, os_build, compiler`, and when I set `arch_build` to `x86`, `MSBuild` will always build using x86_64 architecture.